crates/solver/src/solver.rs

```diff
@@ -224,22 +225,31 @@ pub enum SolverType {
 #[derive(Clone)]
 pub enum SolverAccountArg {
     PrivateKey(PrivateKey),
+    Kms(Arn),
     Address(H160),
 }
 
 impl Debug for SolverAccountArg {
     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
         match self {
             SolverAccountArg::PrivateKey(k) => write!(f, "PrivateKey({:?})", k.public_address()),
+            SolverAccountArg::Kms(key_id) => write!(f, "KMS({:?})", key_id),
             SolverAccountArg::Address(a) => write!(f, "Address({a:?})"),
         }
     }
 }
 
 impl SolverAccountArg {
-    pub fn into_account(self, chain_id: u64) -> Account {
+    pub async fn into_account(self, chain_id: u64) -> Account {
         match self {
             SolverAccountArg::PrivateKey(key) => Account::Offline(key, Some(chain_id)),
+            SolverAccountArg::Kms(key_id) => {
+                let config = ethcontract::aws_config::load_from_env().await;
```

Comment:
`load_from_env` doesn't document what it does which makes it hard to reason about. I prefer using the builder methods on the aws config explicitly with our own explicit configuration that comes from our normal arguments struct.

Reply:
Even if they get "injected" into our pods, I'd like to know what the names of those environment variables are and handle them explicitly in the way I described.

> So the only way to load them is from the AWS defined env variables.

I don't think that's true.

Reply:
Waiting for @ahhda and @mateipopa to give more insight, but I believe things have changed (or are intended to change) from the way they may have been done for the s3 upload.

If I understand correctly we now want to use IRSA/OICD based account authentication (cf. a somewhat easy to read article [here](https://medium.com/@ankit.wal/the-how-of-iam-roles-for-service-accounts-irsa-on-aws-eks-3d76badb8942))

So while we could probably load the credentials via explicit paths it would likely have to be different from local testing where we use session token authentication (and `load_from_env` basically abstracts the different ways of authenticating AWS services away from us). I see your point about this being a bit of a black magic box though.

Reply:
> Can anyone confirm whether that is true?

This was my understanding from @ahhda and @mateipopa as well. Also, "from_env" doesn't mean "from environment variables", but really "from the current system environment". Specifically, credentials [get loaded from data in this order](https://docs.rs/aws-config/latest/aws_config/default_provider/credentials/struct.DefaultCredentialsChain.html):

> - Environment variables: [EnvironmentVariableCredentialsProvider](https://docs.rs/aws-config/latest/aws_config/environment/credentials/struct.EnvironmentVariableCredentialsProvider.html)
> - Shared config (~/.aws/config, ~/.aws/credentials): [SharedConfigCredentialsProvider](https://docs.rs/aws-config/latest/aws_config/profile/struct.ProfileFileCredentialsProvider.html)
> - [Web Identity Tokens](https://docs.rs/aws-config/latest/aws_config/web_identity_token/index.html)
> - ECS (IAM Roles for Tasks) & General HTTP credentials: [ecs](https://docs.rs/aws-config/latest/aws_config/ecs/index.html)
> - [EC2 IMDSv2](https://docs.rs/aws-config/latest/aws_config/imds/index.html)

In theory, `aws_config::load_from_env()` should Just Work™ when called from within an EKS pod (which is what we want).

If we need to, we can add manual configuration knobs in the services for this, but for now I don't think it is necessary.

Reply:
Yes, @nlordell and @fleupold your understanding is correct.

We're no longer required that you explicitly mention each AWS env variable as the credentials are injected into the pod at runtime and use session tokens. `load_from_env` should take care of this for you.

@vkgnosis The approach used for S3 solver instance uploading was an old approach requiring static AWS credentials. We will be updating that to OIDC IRSA as well.